File: java-vertx-web/src/main/java/org/openapitools/vertxweb/server/model/RequestTransportation.java
package org.openapitools.vertxweb.server.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonValue;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class RequestTransportation   {
  


  public enum TypeEnum {
    CYCLING("cycling"),
    DRIVING("driving"),
    DRIVING_TRAIN("driving+train"),
    PUBLIC_TRANSPORT("public_transport"),
    WALKING("walking"),
    COACH("coach"),
    BUS("bus"),
    TRAIN("train"),
    FERRY("ferry"),
    DRIVING_FERRY("driving+ferry"),
    CYCLING_FERRY("cycling+ferry");

    private String value;

    TypeEnum(String value) {
      this.value = value;
    }

    @Override
    @JsonValue
    public String toString() {
      return value;
    }
  }

  private TypeEnum type;
  private Boolean disableBorderCrossing;
  private Integer ptChangeDelay;
  private Integer walkingTime;
  private Integer drivingTimeToStation;
  private Integer cyclingTimeToStation;
  private Integer parkingTime;
  private Integer boardingTime;

  public RequestTransportation () {

  }

  public RequestTransportation (TypeEnum type, Boolean disableBorderCrossing, Integer ptChangeDelay, Integer walkingTime, Integer drivingTimeToStation, Integer cyclingTimeToStation, Integer parkingTime, Integer boardingTime) {
    this.type = type;
    this.disableBorderCrossing = disableBorderCrossing;
    this.ptChangeDelay = ptChangeDelay;
    this.walkingTime = walkingTime;
    this.drivingTimeToStation = drivingTimeToStation;
    this.cyclingTimeToStation = cyclingTimeToStation;
    this.parkingTime = parkingTime;
    this.boardingTime = boardingTime;
  }

    
  @JsonProperty("type")
  public TypeEnum getType() {
    return type;
  }
  public void setType(TypeEnum type) {
    this.type = type;
  }

    
  @JsonProperty("disable_border_crossing")
  public Boolean getDisableBorderCrossing() {
    return disableBorderCrossing;
  }
  public void setDisableBorderCrossing(Boolean disableBorderCrossing) {
    this.disableBorderCrossing = disableBorderCrossing;
  }

    
  @JsonProperty("pt_change_delay")
  public Integer getPtChangeDelay() {
    return ptChangeDelay;
  }
  public void setPtChangeDelay(Integer ptChangeDelay) {
    this.ptChangeDelay = ptChangeDelay;
  }

    
  @JsonProperty("walking_time")
  public Integer getWalkingTime() {
    return walkingTime;
  }
  public void setWalkingTime(Integer walkingTime) {
    this.walkingTime = walkingTime;
  }

    
  @JsonProperty("driving_time_to_station")
  public Integer getDrivingTimeToStation() {
    return drivingTimeToStation;
  }
  public void setDrivingTimeToStation(Integer drivingTimeToStation) {
    this.drivingTimeToStation = drivingTimeToStation;
  }

    
  @JsonProperty("cycling_time_to_station")
  public Integer getCyclingTimeToStation() {
    return cyclingTimeToStation;
  }
  public void setCyclingTimeToStation(Integer cyclingTimeToStation) {
    this.cyclingTimeToStation = cyclingTimeToStation;
  }

    
  @JsonProperty("parking_time")
  public Integer getParkingTime() {
    return parkingTime;
  }
  public void setParkingTime(Integer parkingTime) {
    this.parkingTime = parkingTime;
  }

    
  @JsonProperty("boarding_time")
  public Integer getBoardingTime() {
    return boardingTime;
  }
  public void setBoardingTime(Integer boardingTime) {
    this.boardingTime = boardingTime;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    RequestTransportation requestTransportation = (RequestTransportation) o;
    return Objects.equals(type, requestTransportation.type) &&
        Objects.equals(disableBorderCrossing, requestTransportation.disableBorderCrossing) &&
        Objects.equals(ptChangeDelay, requestTransportation.ptChangeDelay) &&
        Objects.equals(walkingTime, requestTransportation.walkingTime) &&
        Objects.equals(drivingTimeToStation, requestTransportation.drivingTimeToStation) &&
        Objects.equals(cyclingTimeToStation, requestTransportation.cyclingTimeToStation) &&
        Objects.equals(parkingTime, requestTransportation.parkingTime) &&
        Objects.equals(boardingTime, requestTransportation.boardingTime);
  }

  @Override
  public int hashCode() {
    return Objects.hash(type, disableBorderCrossing, ptChangeDelay, walkingTime, drivingTimeToStation, cyclingTimeToStation, parkingTime, boardingTime);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class RequestTransportation {\n");
    
    sb.append("    type: ").append(toIndentedString(type)).append("\n");
    sb.append("    disableBorderCrossing: ").append(toIndentedString(disableBorderCrossing)).append("\n");
    sb.append("    ptChangeDelay: ").append(toIndentedString(ptChangeDelay)).append("\n");
    sb.append("    walkingTime: ").append(toIndentedString(walkingTime)).append("\n");
    sb.append("    drivingTimeToStation: ").append(toIndentedString(drivingTimeToStation)).append("\n");
    sb.append("    cyclingTimeToStation: ").append(toIndentedString(cyclingTimeToStation)).append("\n");
    sb.append("    parkingTime: ").append(toIndentedString(parkingTime)).append("\n");
    sb.append("    boardingTime: ").append(toIndentedString(boardingTime)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: kotlin-spring/src/main/kotlin/org/openapitools/model/RequestTransportation.kt
package org.openapitools.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonValue
import javax.validation.constraints.DecimalMax
import javax.validation.constraints.DecimalMin
import javax.validation.constraints.Max
import javax.validation.constraints.Min
import javax.validation.constraints.NotNull
import javax.validation.constraints.Pattern
import javax.validation.constraints.Size

/**
 * 
 * @param type 
 * @param ptChangeDelay 
 * @param walkingTime 
 * @param drivingTimeToStation 
 * @param parkingTime 
 * @param boardingTime 
 */
data class RequestTransportation(

    @get:NotNull 
    @field:JsonProperty("type") val type: RequestTransportation.Type,

    @field:JsonProperty("pt_change_delay") val ptChangeDelay: kotlin.Int? = null,

    @field:JsonProperty("walking_time") val walkingTime: kotlin.Int? = null,

    @field:JsonProperty("driving_time_to_station") val drivingTimeToStation: kotlin.Int? = null,

    @field:JsonProperty("parking_time") val parkingTime: kotlin.Int? = null,

    @field:JsonProperty("boarding_time") val boardingTime: kotlin.Int? = null
) {

    /**
    * 
    * Values: cycling,driving,drivingPlusTrain,publicTransport,walking,coach,bus,train,ferry,drivingPlusFerry,cyclingPlusFerry
    */
    enum class Type(val value: kotlin.String) {
    
        @JsonProperty("cycling") cycling("cycling"),
    
        @JsonProperty("driving") driving("driving"),
    
        @JsonProperty("driving+train") drivingPlusTrain("driving+train"),
    
        @JsonProperty("public_transport") publicTransport("public_transport"),
    
        @JsonProperty("walking") walking("walking"),
    
        @JsonProperty("coach") coach("coach"),
    
        @JsonProperty("bus") bus("bus"),
    
        @JsonProperty("train") train("train"),
    
        @JsonProperty("ferry") ferry("ferry"),
    
        @JsonProperty("driving+ferry") drivingPlusFerry("driving+ferry"),
    
        @JsonProperty("cycling+ferry") cyclingPlusFerry("cycling+ferry");
    
    }

}

